### DISCUSSION TITLE
Adding an embedded JRE (JustJ) using 'none' executionEnvironment results in Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))

### DISCUSSION THREAD

**Discussion by tim-vbn** [eclipse-tycho/tycho#2677] [Q&A]
I'm trying to add an embedded JRE to our feature-based product but I'm having some trouble.

An example of our Tycho setup can be found here: https://github.com/tim-vbn/indoc

The com.indoc bundle fails to build with the following:

`[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0:validate-classpath (default-validate-classpath) on project com.indoc: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0:validate-classpath failed: org.osgi.framework.BundleException: Bundle com.indoc cannot be resolved:com.indoc [1]
[ERROR]   Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=17))"`

I'm using `<executionEnvironment>none</executionEnvironment>` in the target-platform-configuration based on the latest information I can find.

This bundle has `Bundle-RequiredExecutionEnvironment: JavaSE-17` and if I remove this I get a different error:

`[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0:compile (default-compile) on project com.indoc: Execution default-compile of goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0:compile failed: The 'none' execution environment shouldn't be dereferenced for compilation -> [Help 1]`

I'm not sure what else to try.  Any help is appreciated.

**Comment by tim-vbn** [comment]
Looking through https://eclipse.dev/justj/?page=documentation I ended up trying

`<executionEnvironment>org.eclipse.justj.openjdk.hotspot.jre.full-17</executionEnvironment>`

and the build succeeds now.  Is this the correct way to do it now?

**Comment by laeubi** [comment]
If you use `<executionEnvironment>none</executionEnvironment>` you should **only** activate this for the product module build, not for all modules, that's what the error says, if `<executionEnvironment>org.eclipse.justj.openjdk.hotspot.jre.full-17</executionEnvironment>` works you can use that of course as well.
